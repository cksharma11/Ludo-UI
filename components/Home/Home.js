import React from 'react';
import PropType from 'prop-types';
import Header from '../Header/Header';
import HomeStyles from './Home.style';
import labels from '../../config/labels/labels';
import globalStyles from '../GlobalStyle';

const Home = ({ title, createGameLink, joinGameLink, loadGameLink }) => {
  return (
    <div className="main-container">
      <Header title={title} />
      <div className="options">
        <a href={createGameLink}>{labels.CREATE_GAME}</a>
        <a href={joinGameLink}>{labels.JOIN_GAME}</a>
        <a href={loadGameLink}>{labels.LOAD_GAME}</a>
      </div>
      <style jsx>{HomeStyles}</style>
      <style jsx>{globalStyles}</style>
    </div>
  );
};

Home.propTypes = {
  title: PropType.string.isRequired,
  createGameLink: PropType.string.isRequired,
  joinGameLink: PropType.string.isRequired,
  loadGameLink: PropType.string.isRequired
};

export default Home;
