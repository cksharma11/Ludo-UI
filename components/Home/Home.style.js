import css from 'styled-jsx/css';
import colors from '../../config/styles/colors';

const HomeStyles = css`
  html,
  body {
    width: 100%;
    height: 100%;
  }

  .main-container {
    font-family: fantasy;
    display: flex;
    flex-direction: column;
  }

  .options {
    font-family: fantasy;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background: ${colors.gray.shade2};
    padding: 100px;
  }

  a {
    border-radius: 2px;
    font-family: fantasy;
    width: 200px;
    height: 30px;
    border: 1px solid black;
    color: ${colors.gray.shade1};
    text-align: center;
    line-height: 2;
    text-decoration: none;
    padding: 5px;
    margin: 5px;
  }
`;

export default HomeStyles;
