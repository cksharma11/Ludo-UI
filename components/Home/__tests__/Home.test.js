import React from 'react';
import { shallow } from 'enzyme';
import Home from '../Home';

describe('Home', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <Home title="title" createGameLink="#" joinGameLink="#" />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
