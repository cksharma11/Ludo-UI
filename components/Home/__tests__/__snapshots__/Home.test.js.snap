// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render properly 1`] = `
"<div className=\\"jsx-1535714850 main-container\\">
  <Header title=\\"title\\" />
  <div className=\\"jsx-1535714850 options\\">
    <a href=\\"#\\" className=\\"jsx-1535714850\\">
      Create Game
    </a>
    <a href=\\"#\\" className=\\"jsx-1535714850\\">
      Join Game
    </a>
    <a href={[undefined]} className=\\"jsx-1535714850\\">
      Load Game
    </a>
  </div>
  <JSXStyle id=\\"1535714850\\">
    .
    m
    a
    i
    n
    -
    c
    o
    n
    t
    a
    i
    n
    e
    r
    .
    j
    s
    x
    -
    1
    5
    3
    5
    7
    1
    4
    8
    5
    0
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    8
    5
    v
    h
    p
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    }
    .
    o
    p
    t
    i
    o
    n
    s
    .
    j
    s
    x
    -
    1
    5
    3
    5
    7
    1
    4
    8
    5
    0
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    6
    0
    0
    p
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    a
    l
    i
    g
    n
    -
    i
    t
    e
    m
    s
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    a
    l
    i
    g
    n
    -
    i
    t
    e
    m
    s
    :
    c
    e
    n
    t
    e
    r
    ;
    p
    a
    d
    d
    i
    n
    g
    :
    5
    0
    p
    x
    ;
    }
    a
    .
    j
    s
    x
    -
    1
    5
    3
    5
    7
    1
    4
    8
    5
    0
    {
    b
    o
    r
    d
    e
    r
    -
    r
    a
    d
    i
    u
    s
    :
    2
    p
    x
    ;
    w
    i
    d
    t
    h
    :
    2
    0
    0
    p
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    3
    0
    p
    x
    ;
    b
    o
    r
    d
    e
    r
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    2
    2
    2
    8
    3
    1
    ;
    c
    o
    l
    o
    r
    :
    b
    l
    a
    c
    k
    ;
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    l
    i
    n
    e
    -
    h
    e
    i
    g
    h
    t
    :
    2
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    t
    e
    x
    t
    -
    d
    e
    c
    o
    r
    a
    t
    i
    o
    n
    :
    n
    o
    n
    e
    ;
    t
    e
    x
    t
    -
    d
    e
    c
    o
    r
    a
    t
    i
    o
    n
    :
    n
    o
    n
    e
    ;
    p
    a
    d
    d
    i
    n
    g
    :
    5
    p
    x
    ;
    m
    a
    r
    g
    i
    n
    :
    5
    p
    x
    ;
    }
    a
    .
    j
    s
    x
    -
    1
    5
    3
    5
    7
    1
    4
    8
    5
    0
    :
    h
    o
    v
    e
    r
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    w
    h
    i
    t
    e
    ;
    }
  </JSXStyle>
</div>"
`;
